cmd/podman/cp.go

```diff
@@ -253,11 +272,58 @@ func copy(src, destPath, dest string, idMappingOpts storage.IDMappingOptions, ch
 	if srcfi.IsDir() {
 
 		logrus.Debugf("copying %q to %q", srcPath+string(os.PathSeparator)+"*", dest+string(os.PathSeparator)+"*")
-		if err = copyWithTar(srcPath, destPath); err != nil {
-			return errors.Wrapf(err, "error copying %q to %q", srcPath, dest)
+		if len(copyOpts.excludes) == 0 {
+			if err = copyWithTar(srcPath, destPath); err != nil {
+				return errors.Wrapf(err, "error copying %q to %q", srcPath, dest)
+			}
+			return nil
+		}
+		err := filepath.Walk(srcAbsPath, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
+			}
+			for _, exclude := range copyOpts.excludes {
+				match, err := filepath.Match(filepath.Clean(exclude.ExcludePath), filepath.Clean(path))
+				if err != nil {
+					return err
+				}
+				if !match {
+					continue
+				}
+				if exclude.IsExcluded {
+					return nil
+				}
+				break
```

Comment:
why `break` here?  Don't we need to check all the exclude rules?

Reply:
if it's not match, I have `continue` in line 295

Reply:
shouldn't we consider the last match though?

Something like:
```
a/b
!a/b/c
```
Wouldn't we skip `a/b/c` in this way?

Reply:
yes, @giuseppe , needs review :male_detective: